Tevling/Pages/Statistics.razor.cs

```diff
@@ -122,35 +153,47 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
 
     private async Task UpdateMeasurementData()
     {
+        DateTimeOffset startDate = TimePeriod switch
+        {
+            TimePeriod.Months => DateTimeOffset.Now.AddMonths(-NumberOfPeriodsToReview + 1).ToFirstOfTheMonth(),
+            TimePeriod.Weeks => DateTimeOffset.Now.AddDays(-NumberOfPeriodsToReview * 7),
```

Comment:
Do we need a similar `ToStartOfTheWeek` extension method here to make sure we include the full week?